### ISSUE TITLE
[Help Request] Role cannot be empty when authentication with the PLAIN mechanism

### ISSUE THREAD

**Issue by KamenRiderKuuga** [streamnative/kop#1929]
**Problem**

When authentication with the PLAIN mechanism, got the error msg: Role cannot be empty

**To Reproduce**

The producer code:

```java
package org.example;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.IntegerSerializer;
import org.apache.kafka.common.serialization.StringSerializer;

import java.util.Properties;

public class Main {
    public static void main(String[] args) {
        Properties props = new Properties();
        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "127.0.0.1:" + 9092);
        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);

        String jaasTemplate = "org.apache.kafka.common.security.plain.PlainLoginModule "
                + "required username=\"%s\" password=\"%s\";";
        String jaasCfg = String.format(jaasTemplate, "public/default", "token:" + "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjpbXSwiaWF0IjoxNjg3NjMwMjY5LCJleHAiOjE4NDU0NzUxOTksImF1ZCI6IiIsImlzcyI6IiIsInN1YiI6IiJ9.bVKykCNNg4eh3FiiUf69ZkGkyMSN_OrAOJUHS9bP5fg");
        props.put("sasl.jaas.config", jaasCfg);
        props.put("security.protocol", "SASL_PLAINTEXT");
        props.put("sasl.mechanism", "PLAIN");
        var producer = new KafkaProducer<>(props);
        producer.send(new ProducerRecord<>("test", "kop-demo"));
    }
}
```

**Screenshots**
![image](https://github.com/streamnative/kop/assets/39508075/c3d701d8-e2b2-4661-8899-87793f8c64ad)

**Configurations**

I'm using a standalone broker. (`streamnative/sn-pulsar:2.10.2.4`)

There are all the configurations under the `conf` folder:
[standalone.conf](https://gist.github.com/KamenRiderKuuga/a530e724c292359282882758214ccdd3#file-standalone-conf)
[client.conf](https://gist.github.com/KamenRiderKuuga/a530e724c292359282882758214ccdd3#file-client-conf)
[proxy.conf](https://gist.github.com/KamenRiderKuuga/a530e724c292359282882758214ccdd3#file-proxy-conf)
[broker.conf(if needed)](https://gist.github.com/KamenRiderKuuga/a530e724c292359282882758214ccdd3#file-broker-conf)

**PS**

The code I can produce messages:
```python
from pulsar import Client, AuthenticationToken

# create an authentication token object with the token string
auth = AuthenticationToken('eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjpbXSwiaWF0IjoxNjg3NjMwMjY5LCJleHAiOjE4NDU0NzUxOTksImF1ZCI6IiIsImlzcyI6IiIsInN1YiI6IiJ9.bVKykCNNg4eh3FiiUf69ZkGkyMSN_OrAOJUHS9bP5fg')

# create a pulsar client with the authentication token object
client = Client('pulsar://127.0.0.1:6650', authentication=auth)

producer = client.create_producer(
    topic='persistent://public/default/test',
)

producer.send('demo-msg'.encode('utf-8'))
client.close()
```

**Comment by BewareMyPower** [comment]
I see you set up a Pulsar proxy, KoP might not work well with the proxy.

Could you also upload the logs from broker?

**Comment by BewareMyPower** [comment]
I just rechecked again, it's expected that the token you used did not work.

<img width="1256" alt="image" src="https://github.com/streamnative/kop/assets/18204803/937cf9d6-1c2b-46d5-b7e2-7f1c165c995f">

The "sub" field is the role, which is empty.

I generated a token following the guide [here](https://pulsar.apache.org/docs/3.0.x/security-jwt/#generate-tokens) and you can see the "sub" field is "test_user"

<img width="1196" alt="image" src="https://github.com/streamnative/kop/assets/18204803/8d9ad8bd-c918-4659-a119-a4b92a4f7792">

I think the issue is at the Pulsar side. It seems that Pulsar can work with an empty role when `authorizationEnabled` is false

**Comment by KamenRiderKuuga** [comment]
Nice catch! Fast and accurate!

I did overlook this point, did not carefully compare the guide and the "sub" field in the Token I generated myself, and I think I can use the token in the native pulsar client because I configured `anonymousUserRole=admin` in the `standalone.conf` file.

I think I should raise an issue about the "sub" field in the token to pulsar, because they don't seem to describe what rules need to be followed when generating token by self

A great help, thanks!
